### ISSUE TITLE
Sound Classifier regression with version 6.2.1 over 5.7.1

### ISSUE THREAD

**Issue by rplom** [apple/turicreate#3173]
Using the very same training set the accuracy of version 6.2.1 has regression to being extremely inaccurate.  The second probabilities are often closer to the reality than the primary.

**Comparisons**
turicreate version:5.7.1
mxnet version:1.4.0
numpy version:1.17.2
tensorflow version: 1.13.1

Training a custom neural network -
+-------------------------+-------------------------+-------------------------+-------------------------+
| Iteration               | Training Accuracy       | Validation Accuracy (%) | Elapsed Time            |
+-------------------------+-------------------------+-------------------------+-------------------------+
| 1                       | 0.795                   | 0.724                   | 4393.301                |
+-------------------------+-------------------------+-------------------------+-------------------------+
| 2                       | 0.875                   | 0.757                   | 5949.175                |
+-------------------------+-------------------------+-------------------------+-------------------------+
| 3                       | 0.919                   | 0.771                   | 7498.924                |
+-------------------------+-------------------------+-------------------------+-------------------------+
| 4                       | 0.938                   | 0.771                   | 9049.218                |
+-------------------------+-------------------------+-------------------------+-------------------------+
| 5                       | 0.957                   | 0.785                   | 10599.760               |
+-------------------------+-------------------------+-------------------------+-------------------------+
| 6                       | 0.965                   | 0.783                   | 12151.323               |
+-------------------------+-------------------------+-------------------------+-------------------------+
| 7                       | 0.971                   | 0.788                   | 13704.714               |
+-------------------------+-------------------------+-------------------------+-------------------------+
| 8                       | 0.975                   | 0.786                   | 15260.150               |
+-------------------------+-------------------------+-------------------------+-------------------------+
| 9                       | 0.979                   | 0.790                   | 16817.287               |
+-------------------------+-------------------------+-------------------------+-------------------------+
| 10                      | 0.983                   | 0.793                   | 18446.920               |
+-------------------------+-------------------------+-------------------------+-------------------------+

{'accuracy': 0.8381172272655717, 'auc': nan, 'precision': 0.735701130172406, 'recall': 0.6013583834877269, 'f1_score': 0.5993292004549783, 'log_loss': 0.8225835219466209, 'confusion_matrix': Columns:
	target_label	str
	predicted_label	str
	count	int

Rows: 1665

Data:
+-----------+---------------------+--------------------+-----+-------+
| threshold |         fpr         |        tpr         |  p  |   n   |
+-----------+---------------------+--------------------+-----+-------+
|    0.0    |         1.0         |        1.0         | 377 | 18677 |
|   1e-05   | 0.15179097285431278 | 0.986737400530504  | 377 | 18677 |
|   2e-05   | 0.13894094340632865 | 0.9840848806366048 | 377 | 18677 |
|   3e-05   | 0.13267655405043638 | 0.9840848806366048 | 377 | 18677 |
|   4e-05   |  0.1276971676393425 | 0.9840848806366048 | 377 | 18677 |
|   5e-05   | 0.12405632596241366 | 0.9840848806366048 | 377 | 18677 |
|   6e-05   | 0.12159340365155004 | 0.9840848806366048 | 377 | 18677 |
|   7e-05   | 0.11955881565561921 | 0.9840848806366048 | 377 | 18677 |
|   8e-05   | 0.11720297692348879 | 0.9840848806366048 | 377 | 18677 |
|   9e-05   | 0.11634630829362318 | 0.9840848806366048 | 377 | 18677 |
+-----------+---------------------+--------------------+-----+-------+

=====

turicreate version:6.2.1
mxnet version:1.4.0
numpy version:1.17.2
tensorflow version:2.1.0

Training a custom neural network -
+-------------------------+-------------------------+-------------------------+-------------------------+
| Iteration               | Training Accuracy       | Validation Accuracy (%) | Elapsed Time            |
+-------------------------+-------------------------+-------------------------+-------------------------+
| 1                       | 0.026                   | 0.032                   | 2771.757                |
+-------------------------+-------------------------+-------------------------+-------------------------+
| 2                       | 0.026                   | 0.032                   | 3109.931                |
+-------------------------+-------------------------+-------------------------+-------------------------+
| 3                       | 0.026                   | 0.033                   | 3448.102                |
+-------------------------+-------------------------+-------------------------+-------------------------+
| 4                       | 0.026                   | 0.032                   | 3785.379                |
+-------------------------+-------------------------+-------------------------+-------------------------+
| 5                       | 0.029                   | 0.035                   | 4123.296                |
+-------------------------+-------------------------+-------------------------+-------------------------+
| 6                       | 0.026                   | 0.032                   | 4461.627                |
+-------------------------+-------------------------+-------------------------+-------------------------+
| 7                       | 0.025                   | 0.028                   | 4800.159                |
+-------------------------+-------------------------+-------------------------+-------------------------+
| 8                       | 0.033                   | 0.038                   | 5141.574                |
+-------------------------+-------------------------+-------------------------+-------------------------+
| 9                       | 0.028                   | 0.034                   | 5479.688                |
+-------------------------+-------------------------+-------------------------+-------------------------+
| 10                      | 0.029                   | 0.034                   | 5817.748                |
+-------------------------+-------------------------+-------------------------+-------------------------+

{'accuracy': 0.020172163451566066, 'auc': nan, 'precision': 0.3186556317709977, 'recall': 0.004455717547200197, 'f1_score': 0.0018509491982028775, 'log_loss': 6.173932370971101, 'confusion_matrix': Columns:
	target_label	str
	predicted_label	str
	count	int

Rows: 332

Data:
+-----------+--------------------+-----+-----+-------+-------------+
| threshold |        fpr         | tpr |  p  |   n   |    class    |
+-----------+--------------------+-----+-----+-------+-------------+
|    0.0    |        1.0         | 1.0 | 388 | 18886 | Atmospheres |
|   1e-05   | 0.9990469130572911 | 1.0 | 388 | 18886 | Atmospheres |
|   2e-05   | 0.998305623212962  | 1.0 | 388 | 18886 | Atmospheres |
|   3e-05   | 0.9980408768399873 | 1.0 | 388 | 18886 | Atmospheres |
|   4e-05   | 0.9978290797416075 | 1.0 | 388 | 18886 | Atmospheres |
|   5e-05   | 0.997458434819443  | 1.0 | 388 | 18886 | Atmospheres |
|   6e-05   | 0.9972466377210633 | 1.0 | 388 | 18886 | Atmospheres |
|   7e-05   | 0.9968759927988986 | 1.0 | 388 | 18886 | Atmospheres |
|   8e-05   | 0.9964523986021392 | 1.0 | 388 | 18886 | Atmospheres |
|   9e-05   | 0.9963465000529492 | 1.0 | 388 | 18886 | Atmospheres |
+-----------+--------------------+-----+-----+-------+-------------+
[35200352 rows x 6 columns]

**Comment by rplom** [comment]
The time range is 785ms to 45 seconds, it likely falls on a bell curve, I could write a script to find the median if really needed, there’s 100K sounds.

I’m using Catalina latest build and a Mac Pro 2019 

I noticed the new version takes around 2 hours to analyze and old one around 5 hours. Either it's using the MacPro hardware more efficiently or doing less work?

Are there any special characters on labels that are not allowed?  I use ":" in labels

**Comment by TobyRoseman** [comment]
I suspect what might be happening here is that most of the mini batches have examples for only a small number of the classes, which would likely lead to a suboptimal model. We tracking that issue in #2633.

However the difference in training time is unexpected.

There shouldn't be a restrictions on the characters in labels.

**Comment by TobyRoseman** [comment]
@rplom - TuriCreate 6.3 has now been released. It contains a fix for #2633. Please see if this issues is fixed, or at least improved, with TuriCreate 6.3.

**Comment by TobyRoseman** [comment]
I'm going to close this issues due to inactivity. I suspect this issue was fixed in 6.3. Please let us know if that is not the case.
